### ISSUE TITLE
issue: handleSubmit does not execute onSubmit function (or any other funciton) on react native

### ISSUE THREAD

**Issue by massanen7** [react-hook-form/react-hook-form#8369]
### Version Number

7.31.1

### Codesandbox/Expo snack

https://snack.expo.dev/@bluebill1049/react-hook-form-v7---controller

### Steps to reproduce

1. Go to 'https://snack.expo.dev/@bluebill1049/react-hook-form-v7---controller'
2. Click on 'Button' button
3. See that console log of onSubmit function does not execute

### Expected behaviour

onSubmit function should execute, and you should see the console.log output.

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by Moshyfawn** [comment]
That's because you can field errors.
```json
{
  "errors": {
    "firstName": {
      "type": "required",
      "message": "",
      "ref": {
        "name": "firstName"
      }
    },
    "lastName": {
      "type": "required",
      "message": "",
      "ref": {
        "name": "lastName"
      }
    }
  }
}
```

if you want to do something, if the form submission fails, make sure to provide a second `onInvalid` callback to to the `handleSubmit` form method

```js
const onInvalid = (errors) => console.error(errors)

handleSubmit(onSubmit, onInvalid)
```

**Comment by massanen7** [comment]
Nope, already tried that, and does not print anything. Anything I pass to the handleSubmit function, does not execute. I was first playing with my own project example, but now Im just playing in the example of the documentation, and happens exactly the same. Both onSubmit nor onInvalid, does not execute

**Comment by Moshyfawn** [comment]
Hm.. can't reproduce in [the Snack](https://snack.expo.dev/@moshyfawn/react-hook-form-v7---controller)

https://user-images.githubusercontent.com/16290753/168876788-8d8572d6-458c-489b-a4ac-ac6a222f14e3.mp4

**Comment by bluebill1049** [comment]
which example is in the doc? it's working for me: https://snack.expo.dev/@bluebill1049/react-hook-form-v7---controller

**Comment by Moshyfawn** [comment]
> Hm.. can't reproduce in [the Snack](https://snack.expo.dev/@moshyfawn/react-hook-form-v7---controller)
> 
> https://user-images.githubusercontent.com/16290753/168876788-8d8572d6-458c-489b-a4ac-ac6a222f14e3.mp4
> 
> 
> 

Hm, I see a couple of thumbs up on the issue. Can someone confirm it's not working in [this Snack](https://snack.expo.dev/@moshyfawn/react-hook-form-v7---controller)?

**Comment by massanen7** [comment]
https://user-images.githubusercontent.com/3900702/168992275-9d362016-6b90-4fcf-8efa-1cc5fbb9b118.mov

I'm working on typescript, but it all should work pretty much the same. Here is my recording

**Comment by bluebill1049** [comment]
it doesn't matter if it's typescript or not. please try to reproduce the issue in the snack above.

**Comment by massanen7** [comment]
Here you have pretty much the same structure I have. The "EmailInputScreen.tsx" is where I pretty much copy pasted the example of the documentation to play with react-hook-form. It's in a screen rendered in a stackNavigator, you'll see that onSubmit nor onErrors are never called there.

https://snack.expo.dev/@massanen/07c78c
